app/src/main/java/org/sopt/dosopttemplate/ui/signin/SignInViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package org.sopt.dosopttemplate.ui.signin
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import org.sopt.dosopttemplate.domain.model.SignInStatus
+import org.sopt.dosopttemplate.network.ServicePool
+import org.sopt.dosopttemplate.network.dto.req.SignInRequest
+import org.sopt.dosopttemplate.network.dto.res.SignInResponse
+import retrofit2.HttpException
+
+class SignInViewModel : ViewModel() {
+    val signInStatus: MutableStateFlow<SignInStatus?> = MutableStateFlow(null)
+
+    private val _userInfo = MutableStateFlow<SignInResponse?>(null)
+    val userInfo: StateFlow<SignInResponse?> get() = _userInfo
+
+    fun signIn(userName: String, password: String) {
+        viewModelScope.launch {
+            runCatching {
+                ServicePool.authService.signIn(SignInRequest(password, userName))
+            }.onSuccess { response ->
+                _userInfo.value = response
+                signInStatus.value = SignInStatus.SUCCESS
+            }.onFailure { exception ->
+                Log.e("NetworkTest", "error:$exception")
+                if (exception is HttpException && exception.code() == 400) {
+                    signInStatus.value = SignInStatus.FAILURE
+                } else {
+                    signInStatus.value = SignInStatus.SERVER_ERROR
+                }
```

Comment:
```suggestion
 signInStatus.value = if (exception is HttpException && exception.code() == 400) SignInStatus.FAILURE else SignInStatus.SERVER_ERROR
```
요렇게 중복되는 코드를 없애볼 수도 있을 것 같네요!

Reply:
와 깔끔하고 좋은 것 같아요!!!!!